data_prep/ar_bounds.py

```diff
@@ -0,0 +1,874 @@
+    """Find the geographical bounds of the largest atmospheric river object.
+
+    Args:
+        ar_mask: Binary mask where 1 indicates atmospheric river presence.
+        min_area_gridpoints: Minimum area in grid points for valid AR object.
+
+    Returns:
+        left_lon, right_lon, bottom_lat, top_lat bounds and object metadata.
+    """
+    largest_obj = find_largest_ar_object(
+        ar_mask,
+        min_area_gridpoints=min_area_gridpoints,
+    )
+
+    if largest_obj is None:
+        # No valid AR objects found
+        return np.nan, np.nan, np.nan, np.nan, None
+
+    bounds = largest_obj["bounds"]
+    return (
+        bounds["lon_min"],
+        bounds["lon_max"],
+        bounds["lat_min"],
+        bounds["lat_max"],
+        largest_obj,
+    )
+
+
+def create_case_summary_plot(
+    case_id: int,
+    title: str,
+    ivt_data: xr.DataArray,
+    ar_mask: xr.DataArray,
+    peak_time_idx: int,
+    peak_ivt_value: float,
+    ar_bounds: Dict,
+    buffered_bounds: regions.Region,
+    largest_obj_metadata: Optional[Dict] = None,
+) -> None:
+    """Create a summary plot for each AR case showing IVT, mask, and bounds.
+
+    Args:
+        case_id: Case ID number.
+        title: Event title.
+        ivt_data: Integrated vapor transport data.
+        ar_mask: AR mask data.
+        peak_time_idx: Time index of peak IVT.
+        peak_ivt_value: Peak IVT value.
+        ar_bounds: AR bounds dictionary.
+        buffered_bounds: Buffered bounds region.
+        largest_obj_metadata: Metadata about the largest object.
+    """
+    # Get the time dimension name
+    time_dim = "valid_time" if "valid_time" in ivt_data.dims else "time"
+
+    # Extract data at peak time
+    ivt_peak = ivt_data.isel({time_dim: peak_time_idx})
+    ar_peak = ar_mask.isel({time_dim: peak_time_idx})
+    peak_time = ivt_data[time_dim].isel({time_dim: peak_time_idx}).values
+
+    # Create figure with cartopy subplots
+    fig = plt.figure(figsize=(20, 8))
+
+    # Plot 1: IVT at peak time
+    ax1 = plt.subplot(1, 3, 1, projection=ccrs.PlateCarree())
+    ax1.add_feature(cfeature.COASTLINE, linewidth=0.5)
+    ax1.add_feature(cfeature.BORDERS, linewidth=0.3, alpha=0.7)
+    ax1.add_feature(cfeature.OCEAN, color="lightblue", alpha=0.3)
+    ax1.add_feature(cfeature.LAND, color="white", alpha=0.8)
+
+    # Plot IVT with colormap
+    ivt_plot = ivt_peak.plot(
+        ax=ax1,
+        transform=ccrs.PlateCarree(),
+        cmap="viridis",
+        add_colorbar=False,
+        vmin=0,
+        vmax=1000,
+    )
+    plt.colorbar(ivt_plot, ax=ax1, label="IVT (kg/m/s)", shrink=0.8)
+
+    ax1.set_title(
+        f"IVT at Peak Time\n{pd.to_datetime(peak_time).strftime('%Y-%m-%d %H:%M')}"
+    )
+    ax1.set_extent(
+        [
+            float(ivt_peak.longitude.min()) - 5,
+            float(ivt_peak.longitude.max()) + 5,
+            float(ivt_peak.latitude.min()) - 5,
+            float(ivt_peak.latitude.max()) + 5,
+        ],
```

Comment:
Would recommend factoring out the 5 degree offset and providing this as a function argument.